crates/stc_ts_file_analyzer/src/analyzer/stmt/return_type.rs

```diff
@@ -72,10 +74,34 @@ impl Analyzer<'_, '_> {
         debug_assert!(!self.config.is_builtin, "builtin: visit_stmts_for_return should not be called");
 
         let mut unconditional_throw = None;
+        let mut never_loop = false;
         for stmt in stmts {
-            if let RStmt::Throw(throws) = stmt {
-                unconditional_throw = Some(throws.span);
-                break;
+            match stmt {
+                RStmt::Throw(throws) => {
+                    unconditional_throw = Some(throws.span);
+                    break;
+                }
+                RStmt::While(RWhileStmt {
```

Comment:
Is this logic inevitable?

Reply:
I think handling AST from here is a too bad for future maintenance

Reply:
If so, is there a device that detects the code that causes never within the block in the process of obtaining the return type?

Reply:
If possible, it will also be possible to erase the added extra_error.

Reply:
I think we should add a field to `ReturnValues` and modify the value from validator for children nodes

Reply:
I think it's no different Both my solution and fn check_for_infinite_loop

Even in_reachable is not spreading from fn check_for_infinite_loop

Reply:
It's finally spreading.
By the way, why are repeat statements also checked in_conditional?

Reply:
https://github.com/dudykr/stc/blob/main/crates/stc_ts_file_analyzer/src/analyzer/stmt/mod.rs#L54

Reply:
> By the way, why are loop statements also checked in_conditional?

I think it's my mistake

Reply:
Alternatively, we can add field to the `Ctx` like `in_never_returning_function` and use it from the validator or an expression for a statement

Reply:
I think that better solution is spread is_unreachagble
how do you think?

Reply:
I'm not sure what is spreading `in_unreachable`

Reply:
umm.... propagate ?

Reply:
I think it can be better 👍

Reply:
I fixed it

Reply:
Thank you!